beacon-chain/blockchain/process_block.go

```diff
@@ -31,6 +31,10 @@ import (
 	"go.opencensus.io/trace"
 )
 
+// This defines the lower size at which we recycle the BLS changes pool to avoid
+// memory leaks
```

Comment:
This comment is a little unclear, conflating the reasoning for the recycle operation and also making it sound like this is a lower bound, rather than an upper bound. I think you want to say something like "We recycle the BLS changes pool to avoid the backing map growing without bound. The cycling operation is expensive because it copies all elements, so we only do it when the map is smaller than this upper bound."